Public/Src/Utilities/Collections/ConcurrentBigMapEntry.cs

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+
+namespace BuildXL.Utilities.Collections
+{
+    /// <summary>
+    /// This is a key value pair for the concurrent big map.
+    /// </summary>
+    /// <remarks>
+    /// We can't use the built-in KeyValuePair because it does not implement GetHashCode and Equals

```

Comment:
If we can't use `KeyValuePair`, how did we use it until now?

Reply:
With a lot of extra allocations... I had a test case in a prototype where adding lots of entries were showing up that boxing was allocating hundreds of megs of memory since KeyValuePair<T,U> does is not equatable or implement  gethashcode and Equals therefore was boxing inside of the dictionaries. See: https://devblogs.microsoft.com/premier-developer/performance-implications-of-default-struct-equality-in-c/

Reply:
thanks for the clarification